internal/db/read_writer.go

```diff
@@ -388,6 +397,47 @@ func (rw *Db) DeleteItems(ctx context.Context, deleteItems []interface{}, opt ..
 	return rowsDeleted, nil
 }
 
+// BeginTx will start a transaction and return a RW for use in db operations
+func (w *Db) BeginTx(ctx context.Context) (*Db, *dbw.RW, error) {
```

Comment:
I'm guessing these functions were pulled out from the logic in DoTx...would we want the backoff/retry/ticket already redeemed logic? Not saying we do, not sure we don't :-D